pkg/controller/pd_control.go

```diff
@@ -20,15 +20,12 @@ import (
 
 // getPDClientFromService gets the pd client from the TidbCluster
 func getPDClientFromService(pdControl pdapi.PDControlInterface, tc *v1alpha1.TidbCluster) pdapi.PDClient {
-	if tc.IsHeterogeneous() {
-		if len(tc.Spec.ClusterDomain) > 0 {
-			return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.Spec.ClusterDomain, tc.IsTLSClusterEnabled())
-		}
-		return pdControl.GetPDClient(pdapi.Namespace(tc.GetNamespace()), tc.Spec.Cluster.Name, tc.IsTLSClusterEnabled())
-	}
-	if len(tc.Spec.ClusterDomain) > 0 {
-		return pdControl.GetClusterRefPDClient(pdapi.Namespace(tc.GetNamespace()), tc.GetName(), tc.Spec.ClusterDomain, tc.IsTLSClusterEnabled())
+	if tc.HeterogeneousWithoutLocalPD() {
+		// TODO: to support across k8s cluster without local pd
+		// if tls enable tc.Spec.Cluster.Name should be same as tc.Name? because it will query the secret using this name?
```

Comment:
```suggestion
		// if TLS is enabled, tc.Spec.Cluster.Name should be same as tc.Name? Because it will query the secret using the tc.Spec.Cluster.Name in the following code.
```